graphql/util.go

```diff
@@ -1275,3 +1275,44 @@ func getProjectPermissionLevel(projectPermission ProjectPermission, access Acces
 
 	return permission, level, nil
 }
+
+func canModifyAnnotation(ctx context.Context, obj *restModel.APITask) (bool, error) {
+	authUser := gimlet.GetUser(ctx)
+	permissions := gimlet.PermissionOpts{
+		Resource:      *obj.ProjectId,
+		ResourceType:  evergreen.ProjectResourceType,
+		Permission:    evergreen.PermissionAnnotations,
+		RequiredLevel: evergreen.AnnotationsModify.Value,
+	}
+	if authUser.HasPermission(permissions) {
+		return true, nil
+	}
+	if utility.StringSliceContains(evergreen.PatchRequesters, utility.FromStringPtr(obj.Requester)) {
+		p, err := patch.FindOneId(utility.FromStringPtr(obj.Version))
+		if err != nil {
+			return false, InternalServerError.Send(ctx, fmt.Sprintf("error finding patch for task: %s", err.Error()))
+		}
+		if p == nil {
+			return false, InternalServerError.Send(ctx, "patch for task doesn't exist")
+		}
+		if p.Author == authUser.Username() {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func annotationPermissionHelper(ctx context.Context, taskID string, execution *int) (bool, error) {
+	t, err := getTask(ctx, taskID, execution, "")
+	if err != nil {
+		return false, err
+	}
+	canModify, err := canModifyAnnotation(ctx, t)
+	if err != nil {
+		return false, err
+	}
+	if !canModify {
+		return false, InputValidationError.Send(ctx, "Insufficient permission for modifying annotation.")
```

Comment:
I think this should be `Forbidden`:
```suggestion
		return false, Forbidden.Send(ctx, "insufficient permission for modifying annotation")
```